[Feat] Fastlane을 이용한 Play Store 배포 자동화 구축

HI-JIN2 opened the pull request:
## Summary
<!-- 무슨 이유로 코드를 변경했는지 -->
<!-- 테스트 계획 또는 완료 사항 -->

GitHub Actions와 Fastlane을 연동하여 Google Play Store(Internal Test, Production) 배포 프로세스를 자동화했습니다.   
기존의 수동 빌드 및 업로드 과정을 제거하고, 워크플로우 실행 한 번으로 서명된 AAB 빌드부터 Play Store 업로드까지 완료되도록 구성했습니다.


## Describe your changes
<!-- 변경 또는 추가된 코드, 관련 스크린샷 -->

### 일반 개발 워크플로우

``` mermaid
graph TD
    A[기능 개발] --> B[develop 브랜치에 push]
    B --> C{CI 자동 실행}
    C --> D[CI 빌드 ci_build]
```

### Release 배포 워크플로우

``` mermaid 
graph TD
    A[release-notes/v3.2.0.yml 작성] --> B[develop 브랜치에 Push]
    B --> C[Git Tag 생성 및 Push v3.2.0]
    C --> D{GitHub Actions 자동 실행}
    D --> E[버전 추출 3.2.0]
    D --> F[릴리즈 노트 YAML 로드]
    D --> G[AAB 빌드 Production /w Signing]
    D --> H[Play Store 배포 Internal Track]
    D --> I[Slack 알림]
```

## Issue

- Resolves #87 

## To reviewers
<!-- 어떤 위험이나 장애가 발견되었는지 -->
<!-- 어떤 부분에 리뷰어가 집중하면 좋을지 -->


### 1. Fastlane 설정 최적화

`fastlane/Fastfile`

* **경로 문제 해결**
  GitHub Actions 환경에서 Fastlane 실행 시 발생하던 상대 경로(`../`) 문제를 해결하기 위해
  `PROJECT_ROOT`, `APP_DIR` 등 **절대 경로 상수**를 도입했습니다.

* **Gradle 실행 컨텍스트 수정**
  `Dir.chdir(PROJECT_ROOT)`를 적용하여 Gradle 명령어가 항상 **프로젝트 루트 기준**으로 실행되도록 수정했습니다.

* **배포 Lane 구성**

  * `deploy_internal_test`
    내부 테스트용 단순 배포
  * `deploy_release`
    YAML 기반 릴리즈 노트를 포함한 정식 / 테스트 배포

---

### 2. CI/CD 워크플로우 구축

`.github/workflows/fastlane.yml`

* **자동 배포 트리거**
  `v*` (예: `v1.0.0`) 형식의 태그가 푸시되면
  `deploy_release` 레인이 자동 실행되도록 구성했습니다.

* **서명(Signing) 자동화**
  GitHub Secrets (`KEYSTORE_CONTENT`)에 저장된 Keystore 파일을
  런타임에 디코딩하여 빌드 머신에 주입하도록 구현했습니다.

* **환경 변수 주입**
  배포에 필요한 API Key, 서명 비밀번호 등을
  Fastlane에 **보안적으로 안전하게 전달**하도록 구성했습니다.

---

### 3. Gradle 빌드 설정

`app/build.gradle.kts`

* **Release 서명 설정**
  CI 환경에서 주입된 환경 변수(`KEYSTORE_PATH` 등)를 사용하여
  Release 빌드 시 자동으로 서명(Signing)이 적용되도록
  `signingConfigs`를 추가했습니다.

---

## ✅ 검증 결과

* `feat/fastlane` 브랜치에서 **태그 푸시 시뮬레이션**을 통해

  * 빌드 성공
  * Google Play Store **Internal Track 업로드 성공** 확인
* Keystore 주입 및 AAB 서명 **정상 적용 확인 완료**

---

## 🚀 사용 방법 (Merge 후)

### 방법 1. 자동 배포 (권장)

1. `release/*` 브랜치에서 릴리즈 노트(`release-notes/vX.Y.Z.yml`) 작성, versionName, versionCode 올려주세요.   
2. PR이 머지되면 release_tag.yml이 실행되면서 Tag 생성 -> Release 생성 -> AAB 빌드(서명됨) -> Play Store 배포가 한 번에 끊김 없이 진행됩니다. 별도로 태그를 감지해서 또 다른 워크플로우를 실행시킬 필요가 없습니다.

---

### 방법 2. 수동 배포

GitHub Actions 탭에서 **Fastlane CI/CD 워크플로우**를 선택한 후
`Run workflow`를 통해 수동 배포가 가능합니다.

* **Lane**

  * `deploy_internal_test`
  * `deploy_release`
* **Track**

  * `internal`
  * `alpha`
  * `beta`
  * `production`

HI-JIN2 committed:
chore: .gitignore

HI-JIN2 committed:
chore: .gitignore

HI-JIN2 committed:
Merge remote-tracking branch 'origin/feat/fastlane' into feat/fastlane

# Conflicts:
#	.gitignore

HI-JIN2 committed:
feat: fastlane

HI-JIN2 committed:
feat: github release&tag 릴리즈 노트 자동추출로 변경

HI-JIN2 committed:
fix: internal -> production으로 변경

HI-JIN2 committed:
fix: app 폴더가 없을때 추가하도록

HI-JIN2 committed:
fix: fastlane 보안 및 안정성 개선

- FileUtils.mkdir_p로 안전한 디렉토리 생성
- Base64 형식 검증 및 에러 처리 추가
- 서비스 계정 JSON 파일에 0600 권한 설정으로 보안 강화

HI-JIN2 committed:
refactor: fastlane 워크플로우 간소화 및 경로 명확화

- 중복된 환경 변수 설정 단계 제거
- AAB 아티팩트 경로를 명확한 release 경로로 변경

HI-JIN2 committed:
fix: release 워크플로우 안정성 및 보안 개선

- sleep 3 대신 폴링 방식으로 GitHub Release 생성 확인
- jq 명령어 존재 여부 체크 추가
- curl --show-error 옵션으로 에러 노출 방지

HI-JIN2 committed:
feat: 릴리즈 노트 유효성 검증 강화

- RELEASE_NOTES.txt 빈 파일 체크
- 최소 길이(10자) 검증 추가
- Play Store 정책 위반 가능성 경고

HI-JIN2 committed:
feat: 다국어 릴리즈 노트 관리 시스템 도입

- release-notes/ 디렉토리 추가 및 가이드 문서화
- YAML 기반 버전별 릴리즈 노트 관리 (한국어/영어)
- Fastlane 워크플로우에 수동 릴리즈 배포 기능 추가
- deploy_release lane 추가: YAML 파싱 및 다국어 Play Store 배포
- 워크플로우 입력 필드 확장 (version, track 파라미터)
- 샘플 v3.2.0 릴리즈 노트 포함

HI-JIN2 committed:
Merge branch 'develop' into feat/fastlane

HI-JIN2 committed:
feat: 배포 테스트를 위한 임시 워크플로우 추가

HI-JIN2 committed:
fix: Fastlane 실행 위치 문제 해결을 위한 경로 수정

HI-JIN2 committed:
fix: Fastlane Gradle 실행 시 프로젝트 루트에서 실행하도록 수정

HI-JIN2 committed:
fix: Fastfile 절대 경로 사용 및 Gradle 실행 컨텍스트 수정

HI-JIN2 committed:
fix: Release 빌드 서명 설정 추가 및 CI keystore 주입

HI-JIN2 committed:
fix: 워크플로우 YAML 중복 키 오류 수정

HI-JIN2 committed:
release note: 3.2.1

HI-JIN2 committed:
release note: 3.2.1 en

HI-JIN2 committed:
chore: 테스트 워크플로우 삭제 및 프로덕션 워크플로우에 서명 설정 적용

HI-JIN2 committed:
feat: v* 태그 Push 시 자동 배포(deploy_release) 설정 추가

HI-JIN2 committed:
refactor: 코드 리뷰 반영 (보안 및 안정성 개선)

HI-JIN2 committed:
fix: Release Tag 워크플로우에 서명(Signing) 설정 추가

HI-JIN2 committed:
chore: Fastfile 변경 검증을 위해 테스트 워크플로우(test-deploy.yml) 일시 복구

HI-JIN2 committed:
release: versionCode 49

HI-JIN2 committed:
fix: deploy_internal_test 레인에서 릴리즈 노트 지원 (파일 존재 시)

HI-JIN2 committed:
refactor: Fastlane 레인 간 릴리즈 노트 로직 통일 (YAML 우선 지원)

HI-JIN2 committed:
release: versionCode 50

HI-JIN2 committed:
fix: deploy_internal_test 레인에서 upload_params 변수 초기화 누락 수정

HI-JIN2 committed:
refactor: Fastlane 릴리즈 노트 업로드 방식을 metadata_path 사용하여 표준화

HI-JIN2 committed:
fix: deploy_release 레인 로직 복구 및 메타데이터 업로드 흐름 통일

HI-JIN2 committed:
fix: deploy_internal_test 레인 Ruby 구문 오류 수정 (redundant begin 제거)

HI-JIN2 committed:
fix: Play Store 업로드 시 기본 메타데이터 스킵 (타이틀 누락 오류 방지, 릴리즈 노트만 업로드)

HI-JIN2 committed:
chore: 검증 완료된 테스트 워크플로우(test-deploy.yml) 삭제

PeraSite reviewed (approved):
혼자만 몰래몰래 AGENTS.md 만들어서 썼었는데 이제 올라가니 시원하게 클코 딸깍할 수 있겠네용
너무 편하게 배포할 수 있다니 좋습니다!!

HI-JIN2 committed:
refactor: update release workflow for version-only tags

- Remove 'v' prefix from tags in release_tag.yml
- Trigger fastlane workflow on version-only tags (X.Y.Z)
- Disable redundant Play Store deployment in release_tag workflow
- Fix deploy parameter logic in Fastfile
